### ISSUE TITLE
Request for gateway fees

### ISSUE THREAD

**Issue by dxdc** [ingenesis/shopp#3259]
In the SQL database, the fees deducted by each gateway are recorded as 0.00. I'm only familiar with using the Paypal and Stripe setup.

Paypal returns this information as a variable, and Stripe has a fixed fee structure.

Being able to include this information inside of the database/order transaction itself would greatly simplify bookkeeping. I realize this might be a bit complicated since refunds would have to alter this fee structure, but I think it would make a worthy addition (and should be fairly straightforward?) to implement. It looks like it's already captured to some degree but not being directly updated.

**Comment by msigley** [comment]
This would have to be somewhat of a gateway specific feature as alot of gateways do not report this information back. Also most gateways, Paypal included, have volume pricing based on the number of transactions in a month, which would make the recorded value of the gateway fee incorrect. Other gateways, Authorize.net and some Stripe accounts, are simply API accounts tied to a 3rd Party Merchant account, again they would only return the API fees, not the actual merchant fees, which would make the recorded value of the gateway fee incorrect.

**Comment by dxdc** [comment]
This is all true, and I realize it makes the issue complex. Although, I imagine for most gateways it's either 1) a reported value from the API, or 2) a fixed cost + % of sale. Depending on the volume the fixed cost/% would change, and also refunds make that fairly complicated too.

Ultimately, it would just be nice to simplify the accounting and integrate it with Shopp. Since most gateways don't report what the purchases are for (just the amount), reconciling how much was paid in fees is a nuisance.

Maybe a better way to go about this is a payment report by gateway? It could list total # of orders, total # of discounts, total # of products, total # of taxes, total # of shipping. Just brainstorming here... thoughts?

**Comment by ben72** [comment]
@dxdc I used to export the order history from Shopp and use that as the basis for accounting. But as the orders increase it's just too much manual work handling part refunds and changed payment method* etc as Shopp does not support this atm. so what I do now is just use the payment gateways reporting as that gets me the net sum including all refunds and also the fees are there. This is not optimal for accounting but it's the most feasible way for me now.

It would be interesting to hear how others do accounting with Shopp.

* When a payment fails there is no easy way to set it to succeeded in the order when using for instance a manual payment request instead. Or change the payment method to bank transfer (offline payment). Maybe this will improve in 1.4?

**Comment by dxdc** [comment]
Yes...exactly what I would want also. +1!!

I think an improved reporting structure could handle this.

**Comment by ben72** [comment]
Order editing is planned for a coming release. Not sure it will cover what I listed above but hopefully it will at least eventually.

My guess it that not so many use the Shopp order export or any Shopp reports for actual accounting?

**Comment by msigley** [comment]
@ben72 I think most merchants have these types of reports available to them through the admin panel in their gateway. I know Authorize.net provides transaction statistics (chargeback, capture, refund, and void totals) as well as a returns report for detailed views on the returned items.

**Comment by dxdc** [comment]
Yes that's true.. but they don't correlate it to taxes or product ordered unfortunately. It would be nice to have it all in one.

**Comment by ben72** [comment]
I agree it would be great to have it all in one place. I think many merchants have multiple payment gateways which each handles these kinds of things differently so that's another argument for using Shopp as the basis for accounting.

@msigley are you using only the information available in the gateway for accounting? I think you just have one tax rate in the US? In Sweden we have three and we need to specify sales for each of them. That information is usually not available in the gateway so that's another reason why Shopp reports are needed for accounting.

It would be great to have these high level instructions available for newcomers in an easy to read form. I would have needed that in the beginning. What's available is mostly low level tech info.

**Comment by msigley** [comment]
@ben72 I have custom reports built to help us fulfill our tax and royalty requirements that are in our order fulfillment system and in Shopp. I worked with @jonathandavis to incorporate most of my product sales report's features into 1.3's report system.

All that said, the biggest limitation in Shopp's current reports is there is no easy way to handle these type of correlative analysis situations.

I created an issue for general improvements to the reports in Shopp. #3276
